src/Ocelot/Metadata/DownstreamRouteExtensions.cs

```diff
@@ -0,0 +1,165 @@
+        }
+
+        return (T)ConvertTo(typeof(T), metadataValue, downstreamRoute.MetadataOptions,
+            jsonSerializerOptions ?? new JsonSerializerOptions(JsonSerializerDefaults.Web));
+    }
+
+    /// <summary>
+    /// Converting a string value to the target type
+    /// Some custom conversion has been for the following types:
+    /// bool, bool?, string[], numeric types
+    /// otherwise trying to deserialize the value using the JsonSerializer
+    /// </summary>
+    /// <param name="targetType">The target type.</param>
+    /// <param name="value">The string value.</param>
+    /// <param name="metadataOptions">The metadata options, it includes the global configuration.</param>
+    /// <param name="jsonSerializerOptions">If needed, some custom json serializer options.</param>
+    /// <returns>The converted string.</returns>
+    private static object ConvertTo(Type targetType, string value, MetadataOptions metadataOptions,
+        JsonSerializerOptions jsonSerializerOptions)
+    {
+        if (targetType == typeof(string))
+        {
+            return value;
+        }
+
+        if (targetType == typeof(bool))
+        {
+            return TruthyValues.Contains(value.Trim());
+        }
+
+        if (targetType == typeof(bool?))
+        {
+            if (TruthyValues.Contains(value.Trim()))
+            {
+                return true;
+            }
+
+            if (FalsyValues.Contains(value.Trim()))
+            {
+                return false;
+            }
+
+            return null;
+        }
+
+        if (targetType == typeof(string[]))
+        {
+            if (value == null)
+            {
+                return Array.Empty<string>();
+            }
+
+            return value.Split(metadataOptions.Separators, metadataOptions.StringSplitOption)
+                .Select(s => s.Trim(metadataOptions.TrimChars))
+                .Where(s => !string.IsNullOrEmpty(s))
+                .ToArray();
+        }
+
+        return NumericTypes.Contains(targetType)
+            ? ConvertToNumericType(value, targetType, metadataOptions.CurrentCulture, metadataOptions.NumberStyle)
+            : JsonSerializer.Deserialize(value, targetType, jsonSerializerOptions);
+    }
+
+    /// <summary>
+    /// Converting string to the known numeric types
+    /// </summary>
+    /// <param name="value">The number as string.</param>
+    /// <param name="targetType">The target numeric type.</param>
+    /// <param name="provider">The current format provider.</param>
+    /// <param name="numberStyle">The current number style configuration.</param>
+    /// <returns>The parsed string as object of type targetType.</returns>
+    /// <exception cref="NotImplementedException">Exception thrown if the conversion for the type target type can't be found.</exception>
+    private static object ConvertToNumericType(string value, Type targetType, IFormatProvider provider,
+        NumberStyles numberStyle)
```

Comment:
FYI This conversion problems are solved in .NET 👇 
```C#
Convert.ChangeType(value, targetType, provider);
```

Reply:
@raman-m you asked me to remove any reflection... And the `Convert.ChangeType` method might be slower than direct method calls due to the overhead of reflection and type checking...

Reply:
and @raman-m, if you look at the code closely, you can see that `Convert.ChangeType` is just using the parse method of the target type, eg: 
```c#
public static int ToInt32(string? value, IFormatProvider? provider)
{
    if (value == null)
        return 0;
    return int.Parse(value, provider);
}
```